core/src/main/java/nl/nn/adapterframework/ldap/LdapIsMemberOfPipe.java

```diff
@@ -0,0 +1,181 @@
+
+ * 
+ * @author Ricardo van Holst
+ */
+public class LdapIsMemberOfPipe extends LdapQueryPipeBase {
+		
+	private boolean recursiveSearch = true;
+	private String groupDN;
+	private final static String PARAM_TARGET_GROUP_DN = "groupDN";
+	
+	private String thenForwardName = "then";
+	private String elseForwardName = "else";
+	
+	private String exceptionOnInput;
+	private String thenOnInput;
+	
+	@Override
+	public void configure() throws ConfigurationException {
+		if (!ConfigurationUtils.isConfigurationStubbed(getConfigurationClassLoader())) {
+			if (StringUtils.isNotEmpty(getThenOnInput())) {
+				throw new ConfigurationException("thenOnInput only allowed in stub mode");
+			}
+			if (StringUtils.isNotEmpty(getExceptionOnInput())) {
+				throw new ConfigurationException("exceptionOnInput only allowed in stub mode");
+			}
+		}
+		
+		super.configure();
+	}
+	
+	@Override
+	public PipeRunResult doPipeWithException(Message message, IPipeLineSession session) throws PipeRunException {
+		if (message==null) {
+			throw new PipeRunException(this, getLogPrefix(session) + "input is null");
+		}
+		
+		String searchedDN;
+		try {
+			searchedDN = message.asString();
+		} catch (IOException e) {
+			throw new PipeRunException(this, getLogPrefix(session) + "Failure converting input to string", e);
+		}
+		
+		if (StringUtils.isNotEmpty(getThenOnInput())) {
+			if (getThenOnInput().equals(searchedDN)) {
+				return new PipeRunResult(findForward(getThenForwardName()), message);
+			} else {
+				return new PipeRunResult(findForward(getElseForwardName()), message);
+			}
+		}
+		if (StringUtils.isNotEmpty(getExceptionOnInput()) && getExceptionOnInput().equals(searchedDN)) {
+			throw new PipeRunException(this, getLogPrefix(session)+"exceptionOnInput ["+getExceptionOnInput()+"]");
+		}
+		
+		String groupDN_work;
+		ParameterValueList pvl = null;
+		if (getParameterList() != null) {
+			try {
+				pvl = getParameterList().getValues(message, session);
+			} catch (ParameterException e) {
+				throw new PipeRunException(this, getLogPrefix(session) + "exception on extracting parameters", e);
+			}
+		}
+		groupDN_work = getParameterValue(pvl, PARAM_TARGET_GROUP_DN);
+		if (groupDN_work == null) {
+			groupDN_work = getGroupDN();
+		}
+
+		Set<String> memberships;
+		try {
+			if (isRecursiveSearch()) {
+				memberships= ldapClient.searchRecursivelyViaAttributes(searchedDN, getBaseDN(), "memberOf", groupDN_work);
+			} else {
+				memberships= ldapClient.searchObjectForMultiValuedAttribute(searchedDN, getBaseDN(), "memberOf");
+			}
+			
+			if (memberships.contains(groupDN_work)) {
+				return new PipeRunResult(findForward(getThenForwardName()), message);
+			}
+			return new PipeRunResult(findForward(getElseForwardName()), message);
+		} catch (NamingException e) {
+			throw new PipeRunException(this, getLogPrefix(session) + "exception on ldap lookup", e);
+		}
+	}
+	
+	/**
+	 * When stubbing, create empty ldapClient to prevent NPEs
```

Comment:
Waar dan?

Reply:
Bij de start wordt `ldapClient.open()` gedaan, maar dan moet er wel een ldapClient zijn.

Omdat stubbing anders gaat lopen, is deze constructie verder niet nodig en is dus opgeruimd.